src/array/cpu/spmat_op_impl_csr.cc

```diff
@@ -492,10 +492,17 @@ template CSRMatrix CSRSliceRows<kDGLCPU, int64_t>(CSRMatrix, NDArray);
 
 template <DGLDeviceType XPU, typename IdType>
 CSRMatrix CSRSliceMatrix(
-    CSRMatrix csr, runtime::NDArray rows, runtime::NDArray cols) {
+    CSRMatrix csr, runtime::NDArray rows, runtime::NDArray cols,
+    bool relabel_nodes) {
+  int64_t new_nrows_temp = rows->shape[0];
+  int64_t new_ncols_temp = cols->shape[0];
+  if (!relabel_nodes) {
+    new_nrows_temp = csr.num_rows;
+    new_ncols_temp = csr.num_cols;
+  }
   IdHashMap<IdType> hashmap(cols);
-  const int64_t new_nrows = rows->shape[0];
-  const int64_t new_ncols = cols->shape[0];
+  const int64_t new_nrows = new_nrows_temp;
```

Comment:
`num_nrows_temp` is used here for assignment only? If yes, please assign `new_nrows` with lambda or triple expression to remove the `xxx_temp` varaibles

Reply:
OK, I changed to triple expression.